examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,642 @@
+"""
+==================================================================
+Common pitfalls in interpretation of coefficients of linear models
+==================================================================
+
+Linear models describe situations in which the target value is expected to be
+a linear combination of the features (see the :ref:`linear_model` User Guide
+section for a description of a set of linear model methods available in
+scikit-learn).
+Coefficients in multiple linear models represent the relationship between the
+given feature, :math:`X_i` and the target, :math:`y`, assuming that all the
+other features remain constant (`conditional dependence
+<https://en.wikipedia.org/wiki/Conditional_dependence>`_).
+This is different from plotting :math:`X_i` versus :math:`y` and fitting a
+linear relationship: in that case all possible values of the other features are
+taken into account in the estimation (marginal dependence).
+
+This example will provide some hints in interpreting coefficient in linear
+models, pointing at problems that arise when either the linear model is not
+appropriate to describe the dataset, or when features are correlated.
+
+We will use data from the "Current Population Survey" from 1985 to predict
+wage as a function of various features such as experience, age, or education.
+
+.. contents::
+   :local:
+   :depth: 1
+"""
+
+print(__doc__)
+
+import numpy as np
+import scipy as sp
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+#############################################################################
+# The dataset: wages
+# ------------------
+#
+# We fetch the data from `OpenML <http://openml.org/>`_.
+# Note that setting the parameter `as_frame` to True will retrieve the data
+# as a pandas dataframe.
+
+from sklearn.datasets import fetch_openml
+
+survey = fetch_openml(data_id=534, as_frame=True)
+
+##############################################################################
+# Then, we identify features `X` and targets `y`: the column WAGE is our
+# target variable (i.e., the variable which we want to predict).
+#
+X = survey.data[survey.feature_names]
+X.describe(include="all")
```

Comment:
For the dataframe to be nicely render, we should add the following css trick:

https://github.com/scikit-learn/scikit-learn/pull/13227/files#diff-1ddeb568bd409df15775919c69128c75

It will be nicer when looking at the documentation

Reply:
This should be automatically applied to the documentation, because it is in the skl css. I can't understand right now why it's not (the dataframe class comes from bootstrap.css... :( ). Let's wait for the build....

Reply:
...just noticed that the css is not merged yet.... should I add the css to my PR? Or you can wait for #13227 to be merged?

Reply:
I would say to make it now because we already have a dataframe which will be better displayed.